### PR TITLE
Added support for JSON.RESP

### LINKED ISSUE #500: Add support for command `JSON.RESP`
Add support for the `JSON.RESP` command in DiceDB similar to the [`JSON.RESP` command in Redis](https://redis.io/docs/latest/commands/command-list/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by psrvere**
Closes #500

### CODE REVIEW

**Comment by psrvere** [comment]
@JyotinderSingh - Please review this PR.

File: integration_tests/commands/async/jsonresp_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		path        string
+	}{
+		{
+			name:        "print array with mixed types",
+			commands:    []string{"json.set key $ " + arrayAtRoot, "json.resp key $"},
+			expected:    []interface{}{"OK", []interface{}{"[", "dice", int64(10), "10.5", "true", "(nil)"}},
+			assert_type: []string{"equal", "equal"},
+		},
+		{
+			name:        "print nested array with mixed types",
```
Comment (JyotinderSingh) [inline integration_tests/commands/async/jsonresp_test.go:34]: Add a test case to run `JSON.RESP` at the root of a JSON with nested fields. I think it is already covered in the unit tests, let's add it here for completeness.

File: internal/clientio/resp.go [hunk 2]
```diff
@@ -152,17 +152,42 @@ func encodeString(v string) []byte {
...
+	}
+	return []byte(fmt.Sprintf("+%v\r\n", val))
+}
+
 func Encode(value interface{}, isSimple bool) []byte {
 	switch v := value.(type) {
 	case string:
-		if isSimple {
+		// encode as simple strings
+		if isSimple || v == "[" || v == "{" {
```
Comment (JyotinderSingh) [inline internal/clientio/resp.go:171]: `v` will store the name of the type (string, int, int8, etc.). How can it be `[` or `{`?

File: internal/eval/eval.go [hunk 3]
```diff
@@ -3912,6 +3912,81 @@ func evalJSONOBJKEYS(args []string, store *dstore.Store) []byte {
...
+	case []interface{}:
+		resp = append(resp, "[")
+		for _, value := range json {
+			resp = append(resp, parseJSONStructure(value, true)...)
+		}
+		// wrap in another array to offset print
+		if nested {
+			resp = []interface{}{resp}
+		}
+	}
```
Comment (JyotinderSingh) [inline internal/eval/eval.go:3986]: Maybe we should add a default case here to catch any unexpected types. We can append something like `"(unsupported type)"` into the result. Ideally that condition would never be hit.

File: internal/clientio/resp.go [hunk 4]
```diff
@@ -152,17 +152,42 @@ func encodeString(v string) []byte {
...
+		// since encoding/json unmarshals numeric values as floats
+		// we need to check if value is an integer
+		intValue, isInteger := utils.IsFloatToIntPossible(v.(float64))
+		if isInteger {
+			return []byte(fmt.Sprintf(":%d\r\n", intValue))
+		}
+
+		// if it is a float, encode like a string
+		str := strconv.FormatFloat(v.(float64), 'f', -1, 64)
+		return encodeString(str)
```
Comment (JyotinderSingh) [inline internal/clientio/resp.go:188]: Is this change related to this PR? Also, please add unit tests for these changes.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for adding support for `JSON.RESP` command @psrvere! The changes look good overall, have left a few comments.
